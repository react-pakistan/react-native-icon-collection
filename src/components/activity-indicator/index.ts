export {
  ActivityIndicator,
  ActivityIndicatorSize,
} from './activity-indicator';
