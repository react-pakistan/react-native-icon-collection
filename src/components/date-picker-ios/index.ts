export { DatePickerIOS } from './date-picker-ios';
