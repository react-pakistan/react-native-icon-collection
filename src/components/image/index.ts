export {
  IImageProps,
  Image,
  ResizeMethod,
  ResizeMode,
} from './image';
