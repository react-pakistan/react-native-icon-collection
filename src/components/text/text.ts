import { ITheme } from '@react-pakistan/react-commons-collection/theme';
import { ReactText } from 'react';
import { ThemedStyledProps } from 'styled-components';
import styled from 'styled-components/native';

export const Text = styled.Text<IText>`
  color: ${({ color, theme } : ThemedStyledProps<IText, ITheme>) : string => color || theme.colors.cherryRed};
  font-family: ${({ fontFamily, theme } : ThemedStyledProps<IText, ITheme>) : string => fontFamily || theme.typography.fontFamily};
  font-size: ${({ fontSize, theme } : ThemedStyledProps<IText, ITheme>) : ReactText => fontSize || theme.typography.fontSizeMenuItem}px;
  font-weight: ${({ fontWeight, theme } : ThemedStyledProps<IText, ITheme>) : ReactText => fontWeight || theme.typography.fontWeightButtonText};
  letter-spacing: ${({ letterSpacing, theme } : ThemedStyledProps<IText, ITheme>) : string => letterSpacing || theme.typography.letterSpacingButtonText}px;
  line-height: ${({ lineHeight, theme } : ThemedStyledProps<IText, ITheme>) : ReactText => lineHeight || theme.typography.lineHeightButtonText}px;
  text-align: ${({ textAlign, theme } : ThemedStyledProps<IText, ITheme>) : string => textAlign || theme.typography.textAlignButtonText};
  text-transform: ${({ textTransform, theme } : ThemedStyledProps<IText, ITheme>) : string => textTransform || theme.typography.textTransformButtonText};
`;

export interface IText {
  /**
   *
   */
  color? : string;
  /**
   *
   */
  fontFamily? : string;
  /**
   *
   */
  fontSize? : ReactText;
  /**
   *
   */
  fontWeight? : ReactText;
  /**
   *
   */
  letterSpacing? : string;
  /**
   *
   */
  lineHeight? : ReactText;
  /**
   *
   */
  textAlign? : string;
  /**
   *
   */
  textTransform? : string;
}
