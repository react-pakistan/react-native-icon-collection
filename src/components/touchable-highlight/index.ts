export { TouchableHighlight } from './touchable-highlight';
