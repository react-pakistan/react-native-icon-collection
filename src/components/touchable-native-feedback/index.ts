export { TouchableNativeFeedback } from './touchable-native-feedback';
