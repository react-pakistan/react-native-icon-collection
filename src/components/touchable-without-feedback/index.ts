export { TouchableWithoutFeedback } from './touchable-without-feedback';
