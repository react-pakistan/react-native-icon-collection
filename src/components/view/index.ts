export { View, IView } from './view';
