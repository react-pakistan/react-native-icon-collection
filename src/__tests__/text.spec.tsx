import React from 'react';
import renderer from 'react-test-renderer';
import { Text } from '../components/text';
import {
  defaultTextProps,
} from '../../__stories__/story/text.story';

describe('Testing Text component', () => {
  it('matches Text snapshot, flex column', () => {
    const wrapper = renderer.create(
      <Text {...defaultTextProps} />,
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
