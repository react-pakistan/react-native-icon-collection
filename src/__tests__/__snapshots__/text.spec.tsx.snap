// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Text component matches Text snapshot, flex column 1`] = `
<Text
  color="#00D3CA"
  fontFamily="Helvetica"
  fontSize={32}
  fontWeight={600}
  letterSpacing="1.5"
  lineHeight="32"
  style={
    Array [
      Object {
        "color": "#00D3CA",
        "fontFamily": "Helvetica",
        "fontSize": 32,
        "fontWeight": "600",
        "letterSpacing": 1.5,
        "lineHeight": 32,
        "textAlign": "center",
        "textTransform": "uppercase",
      },
    ]
  }
  textAlign="center"
  textTransform="uppercase"
/>
`;
